jdbc/src/test/java/org/jclouds/jdbc/BaseJdbcBlobStoreTest.java

```diff
@@ -0,0 +1,746 @@
+import org.jclouds.blobstore.domain.BlobMetadata;
+import org.jclouds.blobstore.domain.PageSet;
+import org.jclouds.blobstore.domain.StorageMetadata;
+import org.jclouds.blobstore.domain.StorageType;
+import org.jclouds.blobstore.options.GetOptions;
+import org.jclouds.blobstore.options.ListContainerOptions;
+import org.jclouds.http.HttpRequest;
+import org.jclouds.io.Payload;
+import org.jclouds.io.payloads.PhantomPayload;
+import org.jclouds.io.payloads.StringPayload;
+import org.jclouds.util.Closeables2;
+import org.jclouds.util.Strings2;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.google.common.io.BaseEncoding.base16;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jclouds.utils.TestUtils.randomByteSource;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.FileAssert.fail;
+
+public abstract class BaseJdbcBlobStoreTest {
+
+   private static final String CONTAINER_NAME = "test-container";
+   private static final String BLOB_NAME = "jclouds-test-blob";
+   private static final String PROVIDER = "jdbc";
+
+   private final String jpaModuleName;
+
+   private BlobStoreContext context = null;
+   private BlobStore blobStore = null;
+
+   protected BaseJdbcBlobStoreTest(String jpaModuleName) {
+      this.jpaModuleName = jpaModuleName;
+   }
+
+   @BeforeMethod
+   protected void setUp() throws Exception {
+      context = ContextBuilder.newBuilder(PROVIDER)
+            .modules(ImmutableSet.<Module> of(new JpaPersistModule(jpaModuleName)))
+            .build(BlobStoreContext.class);
+      blobStore = context.getBlobStore();
+   }
+
+   @AfterMethod
+   protected void tearDown() throws IOException {
+      context.utils().injector().getInstance(PersistService.class).stop();
+      context.close();
+   }
+
+   @Test
+   public void testCreateContainerDoesNotExist() {
+      assertThat(blobStore.containerExists(CONTAINER_NAME)).isFalse();
+   }
+
+   @Test
+   public void testCreateContainerInLocation() {
+      assertThat(blobStore.createContainerInLocation(null, CONTAINER_NAME)).isTrue();
+      assertThat(blobStore.containerExists(CONTAINER_NAME)).isTrue();
+   }
+
+   @Test
+   public void testDeleteContainer() {
+      assertThat(blobStore.createContainerInLocation(null, CONTAINER_NAME)).isTrue();
+      assertThat(blobStore.containerExists(CONTAINER_NAME)).isTrue();
+      blobStore.deleteContainer(CONTAINER_NAME);
+      assertThat(blobStore.containerExists(CONTAINER_NAME)).isFalse();
+   }
+
+   @Test
+   public void testPutBlob() {
```

Comment:
Yes, there are many putBlob/getBlob tests in the BaseJdbcStorageStrategyTest and BaseJdbcBlobStoreTest classes.